packages/arb-avm-cpp/CMakeLists.txt

```diff
@@ -83,16 +91,58 @@ elseif (${AVM_FUZZER} MATCHES "honggfuzz")
     add_link_options(-fsanitize=address)
 endif ()
 
+if(NOT arb_node_LINKER_LIBS)
+    set(arb_node_LINKER_LIBS "")
+endif()
+
 find_package(Threads REQUIRED)
 find_package(GMP REQUIRED)
 find_package(GMPXX REQUIRED)
 
+set(ALT_MALLOC_FLAGS "-fno-builtin-malloc -fno-builtin-calloc -fno-builtin-realloc -fno-builtin-free")
+
+if(ENABLE_GPROF)
+    message(STATUS "Using GPROF")
+endif()
+
+if (ENABLE_GPROF)
+    find_package(Gperftools)
+    if(GPERFTOOLS_FOUND)
+        message(STATUS "Using GPROF")
+        include_directories(${GPERFTOOLS_INCLUDE_DIR})
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${ALT_MALLOC_FLAGS} -fno-omit-frame-pointer -g -pg")
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${ALT_MALLOC_FLAGS} -fno-omit-frame-pointer -g -pg")
+        set(arb_node_LINKER_LIBS ${arb_node_LINKER_LIBS} ${GPERFTOOLS_PROFILER})
+        set(CMAKE_LINK_LIBRARY_FILE_FLAG "${CMAKE_LINK_LIBRARY_FILE_FLAG} -g -pg")
+        set(ENABLE_JEMALLOC 0)
+        set(ENABLE_TCMALLOC 1)
+    endif(GPERFTOOLS_FOUND)
+endif (ENABLE_GPROF)
+
+if (ENABLE_TCMALLOC)
+    find_package(Gperftools)
+    if(GPERFTOOLS_FOUND)
+        message(STATUS "Using tcmalloc")
+        include_directories(${GPERFTOOLS_INCLUDE_DIR})
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${ALT_MALLOC_FLAGS}")
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${ALT_MALLOC_FLAGS}")
+        set(arb_node_LINKER_LIBS ${arb_node_LINKER_LIBS} ${GPERFTOOLS_TCMALLOC})
+        set(ENABLE_JEMALLOC 0)
+    endif(GPERFTOOLS_FOUND)
+endif (ENABLE_TCMALLOC)
+
 if (ENABLE_JEMALLOC)
+    if(GPERFTOOLS_FOUND)
```

Comment:
Probably should just explicitly fail if ENABLE_TCMALLOC and ENABLE_JEMALLOC are both true

Reply:
:+1: